src/daemon/ca/publishing.rs

```diff
@@ -1279,95 +998,76 @@ impl BasicKeyObjectSet {
     fn add_elements(&self, map: &mut HashMap<RepositoryContact, Vec<PublishElement>>, dflt_repo: &RepositoryContact) {
         let repo = self.old_repo.as_ref().unwrap_or(dflt_repo);
 
-        let base_uri = self.signing_cert.ca_repository();
-        let mft_uri = base_uri.join(self.manifest.name().as_bytes()).unwrap();
-        let crl_uri = base_uri.join(self.crl.name().as_bytes()).unwrap();
+        let crl_uri = self.signing_cert.crl_uri();
+        let mft_uri = self.signing_cert.mft_uri();
 
         let elements = map.entry(repo.clone()).or_insert_with(Vec::new);
-        elements.push(PublishElement::new(Base64::from(&self.manifest.0), mft_uri));
-        elements.push(PublishElement::new(Base64::from(&self.crl.0), crl_uri));
+
+        elements.push(self.manifest.publish_element(mft_uri));
+        elements.push(self.crl.publish_element(crl_uri));
     }
 
     fn create(key: &CertifiedKey, timing: &IssuanceTimingConfig, signer: &KrillSigner) -> KrillResult<Self> {
         let signing_cert = key.incoming_cert().clone();
 
-        let signing_key = signing_cert.subject_public_key_info();
+        let signing_key = signing_cert.key_identifier();
         let issuer = signing_cert.subject().clone();
         let revocations = Revocations::default();
-        let number = 1;
-        let next_update = timing.publish_next();
+        let revision = ObjectSetRevision::create(timing);
 
-        let crl = CrlBuilder::build(signing_key, issuer, &revocations, number, next_update, signer)?;
+        let crl = CrlBuilder::build(signing_key, issuer, &revocations, revision, signer)?;
 
-        let manifest = ManifestBuilder::with_crl_only(&crl)
-            .build_new_mft(&signing_cert, number, signer)
+        let manifest = ManifestBuilder::new(revision)
+            .with_crl_only(&crl)
+            .build_new_mft(&signing_cert, signer)
             .map(|m| m.into())?;
 
-        Ok(BasicKeyObjectSet::new(signing_cert, number, revocations, manifest, crl))
+        Ok(BasicKeyObjectSet::new(
+            signing_cert,
+            revision,
+            revocations,
+            manifest,
+            crl,
+            None,
+        ))
     }
 
     pub fn requires_reissuance(&self, hours: i64) -> bool {
-        Time::now() + Duration::hours(hours) > self.manifest.next_update()
-            || Some(self.signing_cert.uri()) != self.manifest.cert().ca_issuer()
+        Time::now() + Duration::hours(hours) > self.next_update()
```

Comment:
Because just re-issuing the MFT is not enough, and a bit pointless. We should re-issue everything in the rare case that this should happen. I created this issue for it: https://github.com/NLnetLabs/krill/issues/840

Furthermore.. will present and discuss at the IETF.

All this said, it's not the end of the world when the URI pointing to the parent certificate is wrong. RPs accept this.